test_util/aws.py

```diff
@@ -170,30 +195,14 @@ def empty_and_delete_s3_bucket_from_stack(self):
         bucket.delete()
         log.info('Delete successfully triggered for {}'.format(self.stack.stack_name))
 
-    def get_master_ips(self, state_list=None):
-        instances = self.get_group_instances(['MasterServerGroup'], state_list)
-        return instances_to_hosts(instances)
-
-    def get_public_agent_ips(self, state_list=None):
-        instances = self.get_group_instances(['PublicSlaveServerGroup'], state_list)
-        return instances_to_hosts(instances)
+    def get_master_ips(self):
+        return instances_to_hosts(self.get_auto_scaling_instances('MasterServerGroup'))
```

Comment:
For caching, if hosts randomly appear / disappear currently a lot of the code (and all of our tests) I think are currently quite broken, they expect a static list of hosts / no host failures while running.

My biggest concern with caching is if we don't, everytime someone calls get_master_ips or get_public_agent_ips they're implicitly making a whole bunch of AWS API calls, which could get us rate limited.